testsuite/classlibrary/TestTask.sc

```diff
@@ -23,4 +23,40 @@ TestTask : UnitTest {
 		this.assert(task.isPlaying.not, "Task be stoppable even if not running");
 	}
 
+
+	test_play_with_SystemClock {
+		var ok = false;
+		var task = Task { 0.001.wait; ok = true; };
+		task.play(SystemClock);
+		0.002.wait;
+		task.stop;
+		this.assert(ok, "Task plays with SystemClock");
+	}
+
+	test_play_with_TempoClock {
+		var ok = false;
+		var task = Task { 0.001.wait; ok = true; };
+		task.play(TempoClock.default);
+		0.002.wait;
+		task.stop;
+		this.assert(ok, "Task plays with TempoClock.default");
+	}
+
+	test_play_with_AppClock {
+		var ok = false;
+		var task = Task { 0.001.wait; ok = true; };
+		task.play(AppClock);
+		0.002.wait;
+		task.stop;
+		this.assert(ok, "Task plays with AppClock");
+	}
+
+	test_play_with_LinkClock {
+		var ok = false;
+		var task = Task { 0.001.wait; ok = true; };
+		task.play(LinkClock);
```

Comment:
This is not actually testing on a LinkClock.

```
LinkClock.default
-> a TempoClock

Task {
	thisThread.clock.postln;
}.play(LinkClock);

a TempoClock  // printed
```

LinkClock depends on the Ableton Link backend, which is dormant until an instance is created (see `prStart`, calling `_LinkClock_New`). This means, if the code hasn't created an instance, then it isn't using LinkClock.

It may also be necessary to wait a bit longer. *If* there are no other Link peers on the network at the time of running the test, then the Task will awaken *now*, with beats = 0.0 as we would expect from TempoClock. If there are other peers, then the clock time may be adjusted forward or backward to match the Link quantum. This shouldn't be the case in most testing, but you never know.

To cover all the bases, I'd suggest like this:

```
	test_play_with_LinkClock {
		var ok = false;
		var task = Task { 0.001.wait; ok = true; cond.signalAll };
		var clock = LinkClock.new;
		var cond = CondVar.new;
		task.play(LinkClock);
		cond.waitFor(1, { ok });
		task.stop;
		clock.stop;
		this.assert(ok, "Task plays with LinkClock");
	}
```